analysis_options.yaml

```diff
@@ -7,17 +7,18 @@
 # See the configuration guide for more
 # https://github.com/dart-lang/sdk/tree/master/pkg/analyzer#configuring-the-analyzer
 #
-# There are four similar analysis options files in the flutter repos:
+# There are five similar analysis options files in the flutter repo:
 #   - analysis_options.yaml (this file)
+#   - analysis_options_repo.yaml
```

Comment:
It looks like `analysis_options_repo.yaml` was intended to be removed in https://github.com/flutter/flutter/pull/16979.  If so, we should audit any changes that have landed to that file since that PR landed and make sure things haven't gotten out of sync.

/cc @a14n @devoncarew

Reply:
I thought that there was a legitimate reason why there were two almost identical files: according to the comments, one is for flutter analyze to use, and one is for IntelliJ.  I did check and make sure that they hadn't diverged (there is one intentional difference, turning off `public_member_api_docs` for IntelliJ which I think is the reason for the separate files).

Reply:
iirc @devoncarew unified them in that PR

Reply:
Yes, I'd thought I'd deleted one of them in that PR. I had to revert and re-apply it a few weeks later, so it's possible the delete was a casualty.

We should just have one at the top level (I believe `analysis_options.yaml`), and it should have the `public_member_api_docs` lint enabled.

Reply:
OK, I've removed it and updated the comments at the top of the file. I grepped through the repo, and I didn't see any reference to it.